Tests/SwiftDocCTests/Rendering/Markdown/MarkdownOutputTests.swift

```diff
@@ -0,0 +1,971 @@
+            # Snippets
+            
+            Here is an article with some snippets
+            
+            ## Overview
+            
+            @Snippet(path: "MarkdownOutput/SnippetA", slice: "sliceOne")
+            
+            Post snippet content
+            """)
+        
+        let snippetContent = """
+        import Foundation
+        // I am a code snippet
+        
+        // snippet.hide
+        // I am hidden content
+        """
+        
+        let snippet = makeSnippet(pathComponents: ["MarkdownOutput", "SnippetA"], explanation: nil, code: snippetContent)
+        let graph = JSONFile(name: "MarkdownOutput.symbols.json", content: makeSymbolGraph(moduleName: "MarkdownOutput", symbols: [snippet]))
+        
+        let catalog = catalog(files: [articleWithSnippet, graph])
+        let (node, _) = try await markdownOutput(catalog: catalog, path: "SnippetArticle")
+        XCTAssertFalse(node.markdown.contains("// I am hidden content"))
+    }
+    
+    func testSnippetInclusionWithExplanation() async throws {
+        let articleWithSnippet = TextFile(name: "SnippetArticle.md", utf8Content: """
+            # Snippets
+            
+            Here is an article with some snippets
+            
+            ## Overview
+            
+            @Snippet(path: "MarkdownOutput/SnippetA")
+            
+            Post snippet content
+            """)
+        
+        let snippetContent = """
+        import Foundation
+        // I am a code snippet
+        """
+        
+        let explanation = """
+        I am the explanatory text.
+        I am two lines long.
+        """
+        let snippet = makeSnippet(pathComponents: ["MarkdownOutput", "SnippetA"], explanation: explanation, code: snippetContent)
+        let graph = JSONFile(name: "MarkdownOutput.symbols.json", content: makeSymbolGraph(moduleName: "MarkdownOutput", symbols: [snippet]))
+        
+        let catalog = catalog(files: [articleWithSnippet, graph])
+        let (node, _) = try await markdownOutput(catalog: catalog, path: "SnippetArticle")
+        XCTAssert(node.markdown.contains(explanation))
+    }
+      
+    private func makeSnippet(
+        pathComponents: [String],
+        explanation: String?,
+        code: String,
+        slices: [String: Range<Int>] = [:]
+    ) -> SymbolGraph.Symbol {
+        makeSymbol(
+            id: "$snippet__module-name.\(pathComponents.map { $0.lowercased() }.joined(separator: "."))",
+            kind: .snippet,
+            pathComponents: pathComponents,
+            docComment: explanation,
+            otherMixins: [
+                SymbolGraph.Symbol.Snippet(
+                    language: SourceLanguage.swift.id,
+                    lines: code.components(separatedBy: "\n"),
+                    slices: slices
+                )
+            ]
+        )
+    }
+    
+    func testBadlyFormattedTablesCrash() async throws {
```

Comment:
There's quite a lot of table syntax that has no coverage here. Please also add tests that verify:
- different column alignments (see [Aligning content in a column](https://www.swift.org/documentation/docc/adding-tables-of-data#Aligning-content-in-a-column))
- cells spanning more than one _column_ (see [Spanning cells across columns](https://www.swift.org/documentation/docc/adding-tables-of-data#Spanning-cells-across-columns))
- cells spanning more than one _row_ (see [Spanning cells across rows](https://www.swift.org/documentation/docc/adding-tables-of-data#Spanning-cells-across-rows))